File: 20241029/jiyeon/전깃줄.py
import sys
input = sys.stdin.readline

cnt = int(input())
lines = [list(map(int, input().split())) for _ in range(cnt)]
lines.sort(key=lambda x:x[0])

dp = [1 for _ in range(cnt)]

for i in range(1, cnt):
    for j in range(i):
        if(lines[i][1] > lines[j][1]):
            dp[i] = max(dp[i], dp[j]+1)

print(cnt - max(dp))

# 1 8
# 2 2
# 3 9
# 4 1
# 6 4
# 7 6
# 9 7
# 10 10
# 8 2 9 1 4 6 7 10 => 증가하는 가장 긴 수열 구하기

File: 20241029/homin/전깃줄.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});
const input = [];

rl.on("line", (line) => {
  input.push(line);
}).on("close", () => {
  const [n, ...inputs] = input;
  console.log(solution(n, inputs));
});

const solution = (n, inputs) => {
  const line = inputs.map((e) => e.split(" ").map(Number));

  line.sort((a, b) => a[0] - b[0]);

  const newLine = line.map((e) => e[1]);

  const dp = Array(Math.max(n)).fill(1);

  for (let i = 0; i < n; i++) {
    // i는 나자신
    for (let j = 0; j < i; j++) {
      // j는 나보다 작은 애들
      if (newLine[j] < newLine[i]) {
        dp[i] = Math.max(dp[i], dp[j] + 1);
      }
    }
  }
  return n - Math.max(...dp);

  //   정렬 이후에 스택처럼 증가될때만 제거하려고 했는데, 논리가 잘못됨
  //   let current = newLine.pop();
  //   let count = 0;
  //   while (newLine.length) {
  //     const next = newLine.pop();
  //     if (current < next) {
  //       count++;
  //     }
  //     current = next;
  //   }

  //   console.log(count);
};

// 6 8 2 9 5 10 4 1 7 3
